case-lib/pipeline.sh

```diff
@@ -1,11 +1,32 @@
 #!/bin/bash
 
+# This function will evaluate pipeline parameter related shell code generated by
+# - sof-tplgreader.py (for SOF, pipeline parameters dumped from topology)
+# - sof-dump-status.py (for legacy HDA, pipeline paramters dumped from proc)
+# Args: $1: SOF topology path
+#       $2: Pipeline filter in string form
+# Note: for legacy HDA, topology is not present, $1 will be empty.
 func_pipeline_export()
 {
-    # no parameter input the function
-    if [ $# -lt 1 ]; then
-        die "Topology file name is not specified, unable to run command: $SCRIPT_NAME"
+
+    # function parameter check
+    if [ $# -ne 2 ]; then
+        die "Not enough parameters, expect two parameters: topology path and pipeline filter"
     fi
+
+    # For legacy HDA platform, there is no topology, we have to export pipeline
+    # parameters from proc file system.
+    is_sof_used || {
+        filter_str="$2"
+        dlogi "No SOF sound card found, exporting pipeline parameters from proc file system"
+        tmp_pipeline_params=$(mktemp /tmp/pipeline-params.XXXXXXXX)
+        sof-dump-status.py -e "$filter_str" > "$tmp_pipeline_params" ||
+            die "Failed to export pipeline parameters from proc file system"
+        source "$tmp_pipeline_params" || return 1
```

Comment:
Add
```shell
# shellcheck disable=SC1090
```
`pipeline.sh` is one of our few files that are already shelcheck-clean, keep it that way.